tests/components/voice_assistant/test_websocket.py

```diff
@@ -0,0 +1,118 @@
+
+from tests.typing import WebSocketGenerator
+
+
+@pytest.fixture
+async def init_components(hass):
+    """Initialize relevant components with empty configs."""
+    assert await async_setup_component(hass, "homeassistant", {})
+    assert await async_setup_component(hass, "voice_assistant", {})
+
+
+async def test_text_only_pipeline(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator, init_components
+) -> None:
+    """Test events from a pipeline run with text input (no STT/TTS)."""
+    client = await hass_ws_client(hass)
+
+    await client.send_json(
+        {"id": 5, "type": "voice_assistant/run", "intent_input": "Are the lights on?"}
+    )
+
+    # result
+    msg = await client.receive_json()
+    assert msg["success"]
+
+    # run start
+    msg = await client.receive_json()
+    assert msg["event"]["type"] == "run-start"
+
+    # intent
+    msg = await client.receive_json()
+    assert msg["event"]["type"] == "intent-start"
+
+    msg = await client.receive_json()
+    assert msg["event"]["type"] == "intent-finish"
+
+    # run finish
+    msg = await client.receive_json()
+    assert msg["event"]["type"] == "run-finish"
+
+
+async def test_conversation_timeout(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator, init_components
+) -> None:
+    """Test partial pipeline run with conversation agent timeout."""
+    client = await hass_ws_client(hass)
+
+    async def sleepy_converse(*args, **kwargs):
+        await asyncio.sleep(3600)
+
+    with patch(
+        "homeassistant.components.conversation.async_converse", new=sleepy_converse
+    ):
+        await client.send_json(
+            {
+                "id": 5,
+                "type": "voice_assistant/run",
+                "intent_input": "Are the lights on?",
+                "timeout": 0.5,
+            }
+        )
+
+        # result
+        msg = await client.receive_json()
+        assert msg["success"]
+
+        # run start
+        msg = await client.receive_json()
+        assert msg["event"]["type"] == "run-start"
+
+        # intent
+        msg = await client.receive_json()
+        assert msg["event"]["type"] == "intent-start"
+
+        # timeout error
+        msg = await client.receive_json()
+        assert not msg["success"]
+        assert msg["error"]["code"] == "timeout"
+
+
+async def test_pipeline_timeout(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator, init_components
+) -> None:
+    """Test pipeline run with immediate timeout."""
+    client = await hass_ws_client(hass)
+
+    async def sleepy_run(*args, **kwargs):
+        await asyncio.sleep(3600)
+
+    with patch(
+        "homeassistant.components.voice_assistant.pipeline.Pipeline._run",
+        new=sleepy_run,
+    ):
+        await client.send_json(
+            {
+                "id": 5,
+                "type": "voice_assistant/run",
+                "intent_input": "Are the lights on?",
+                "timeout": 0.5,
```

Comment:
can we make the timeout shorter? 
```suggestion
                "timeout": 0.00001,
```

Reply:
It actually loads the default agent sentences, so too short of a timeout might mistakenly fail.